circuit/program/src/data/plaintext/mod.rs

```diff
@@ -27,14 +27,16 @@ mod to_fields;
 
 use crate::{Access, Ciphertext, Identifier, Literal, Visibility};
 use snarkvm_circuit_network::Aleo;
-use snarkvm_circuit_types::{environment::prelude::*, Address, Boolean, Field, Scalar, U16, U8};
+use snarkvm_circuit_types::{environment::prelude::*, Address, Boolean, Field, Scalar, U16, U32, U8};
 
 #[derive(Clone)]
 pub enum Plaintext<A: Aleo> {
     /// A plaintext literal.
     Literal(Literal<A>, OnceCell<Vec<Boolean<A>>>),
     /// A plaintext struct.
     Struct(IndexMap<Identifier<A>, Plaintext<A>>, OnceCell<Vec<Boolean<A>>>),
+    /// A plaintext array.
+    Array(Vec<Plaintext<A>>, OnceCell<Vec<Boolean<A>>>),
```

Comment:
nit: as long as their lengths can't change once created, underlying vectors like these can be converted into boxed slices (with `.into_boxed_slice()`) for tighter capacities and smaller stack sizes; we could do a pass like this later on

Reply:
Thanks, verdict is still out but we want to support dynamic resizing in circuit.